### DISCUSSION TITLE
How to install or sox or how to update with fresh build

### DISCUSSION THREAD

**Discussion by githluapb** [rand256/valetudo#432] [Q&A]
Is there a way to install sox with out using a prebuild image or - in order to keep maps aso. - to flash a new build with restoring existing configuration?

**Comment by githluapb** [comment]
Did some research and found the advice to backup /mnt/data/valetudo:
tar -czf rockrobo.tar.gz
Any thoughts on this? Can somebody confirm that restoring this directory means restoring all settings, maps etc.?

**Comment by rand256** [comment]
First of all, if you have pre-gen3 device and install the image properly (which in fact means - without any factory resets), you won't lose any settings or maps. But keeping a backup is good practice, so basically you need to make a copy of two directories: /mnt/data/valetudo contains all valetudo-related settings and /mnt/data/rockrobo/ contains most device settings.

**Reply by rand256** [reply]
Also you can try download [this file](https://github.com/zvldz/vacuum/blob/master/custom-script/files/sox.tgz) and unpack it into "/" on your device. This is how sox is "installed" into the flashable image.

**Comment by githluapb** [comment]
Thanks a lot, sox works now. I will try to figure out how to update using an image. Is rrr a possible tool to do this conveniently?

**Reply by rand256** [reply]
That depends on the installed firmware. On early versions you can use any tool to flash the image, on later ones - you need either to downgrade the firmware first (by factory reset) or use RE integrated updater found at WebUI -> Settings -> Info.
Manufacturer doesn't like allowing simple installation for third party firmware as you can guess.
